pkg/virt-controller/services/template.go

```diff
@@ -62,6 +70,79 @@ func (t *templateService) RenderLaunchManifest(vm *v1.VM) (*kubev1.Pod, error) {
 	return &pod, nil
 }
 
+func (t *templateService) RenderMigrationJob(vm *v1.VM, sourceNode *kubev1.Node, targetNode *kubev1.Node) (*batch.Job, error) {
+	srcAddr := ""
+	dstAddr := ""
+	for _, addr := range sourceNode.Status.Addresses {
```

Comment:
Actually it is the same logic you wrote, just more compact. We have to go through the whole list to check if we will ever find a `NodeHostName`. If it is found, exist the loop. Otherwise, whenever we see a NodeINternalIP, take it until we find something better. So if no NodeHostName is present, the second one will be chosen ...